# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/SaaDasim05/Flyrank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Provisional lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because I want to investigate whether historical search-performance signals can help prioritize content items that deserve review. The goal is not simply to predict an outcome, but to produce a useful ranked review queue that could help an SEO or content team decide which pages to investigate first. This is a provisional choice and may be refined as I learn more about the available data and evaluate whether the signals are useful and reliable.

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Search question:
Which content items show the strongest evidence of needing a refresh or further investigation, and can historical search-performance signals be used to prioritize a review queue?

Unit of analysis:
The primary unit of analysis will be an individual content item/page.

Decision:
Which pages should the SEO/content team review first?

Action:
The team could investigate the highest-priority pages and decide whether to refresh the content, improve its search presentation, consolidate it, or leave it unchanged after review.

Cost of a wrong recommendation:
A false positive could cause the team to spend time reviewing or updating a page that did not need attention. A false negative could cause a genuinely weak or declining page to be missed, potentially delaying an opportunity to recover search performance.

Why data/ML can help:
The dataset contains multiple search-performance signals, including impressions, clicks, average position, content characteristics, trend information, and query-level signals. Combining these signals may help prioritize pages more consistently than relying on a single rule or manual intuition.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*



In [10]:
import pandas as pd

url = "https://raw.githubusercontent.com/SaaDasim05/Flyrank-ML-Internship/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nColumns:")
print(df.columns.tolist())

Rows: 30000
Columns: 44

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [11]:
# Basic evidence for the Refresh / Content Opportunity lane

print(f"Total content items: {len(df):,}")

print(
    f"Average content age: "
    f"{df['content_age_days'].mean():.1f} days"
)

print(
    f"Average days since last update: "
    f"{df['days_since_last_update'].mean():.1f} days"
)

print("\nTrend direction:")
print(
    df["trend_direction"]
    .value_counts(normalize=True)
    .mul(100)
    .round(1)
)

Total content items: 30,000
Average content age: 256.2 days
Average days since last update: 46.1 days

Trend direction:
trend_direction
down      54.2
stable    19.9
up        14.6
new        7.5
flat       3.8
Name: proportion, dtype: float64


### What I observed

The starter dataset contains **30,000 content items**. The average content age is **256.2 days**, while the average time since the last update is **46.1 days**.

The largest observed trend category is **down**, representing **54.2%** of the content items. This makes content performance changes and potential refresh opportunities worth investigating further.

These numbers do not show that older or recently updated content causes performance changes. They only show that the dataset contains substantial variation in content age, update recency, and observed search-performance direction, providing a useful starting point for the Refresh / Content Opportunity lane.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What I can and cannot claim

I can investigate observed relationships between content characteristics, search-performance signals, and indicators of potential refresh opportunity. I can also evaluate whether a scoring approach can help prioritize content items for human review.

However, I cannot claim that these relationships prove causation or explain Google's ranking algorithm. A high opportunity score would not guarantee that refreshing a page will improve its search performance.

The eventual output should therefore be treated as a prioritization and decision-support tool, not as an automatic recommendation or guaranteed prediction.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


### Self-check

- [x] I selected one of the predefined lanes: Refresh / Content Opportunity Scoring.
- [x] I defined the unit of analysis as an individual content item/page.
- [x] I identified the decision: which pages should be reviewed first?
- [x] I identified possible actions: refresh, investigate, improve, consolidate, or leave unchanged.
- [x] I considered the cost of both false-positive and false-negative recommendations.
- [x] I used real numbers from the starter dataset to support the lane choice.
- [x] I explained why data and ML could help with prioritization.
- [x] I avoided claiming causation or knowledge of Google's ranking algorithm.
- [x] I am treating the lane and question as provisional and will refine them as I learn more about the data.